### DISCUSSION TITLE
Deploy NextJS 13 with appDir on cPanel

### DISCUSSION THREAD

**Discussion by amanvision** [vercel/next.js#44483] [Help] [answered]
To deploy NextJS 13 with appDir enabled, cPanel already have Phusion Passenger installed

1. Uploaded application to the server
2. Created app.js ( startup file ) with the following content: ( copied from this discussion: https://github.com/vercel/next.js/discussions/17289 )
```
const path = require('path');
const nextPath = path.join(__dirname, 'node_modules', '.bin', 'next');
process.argv.length = 1;
process.argv.push(nextPath, 'start');
require(nextPath);
```
3. npm install && npm build ( generated the production build successfully )
4. When i ran the application, it is showing 404 page ( not found ) where as same application working on local machine with all routes.

is there anything needs to be done differently with appDir feature of NextJS 13

Also noted that routes under app directory shows 0B for all routes when building production build

![image](https://user-images.githubusercontent.com/40421653/210182035-6b281ee5-942e-488d-a560-f06679dfca1b.png)

**Comment by colonder** [comment]
@amanvision any luck?

**Reply by amanvision** [reply, marked as answer]
Yes, It's resolved.

1. I used the following code for the entry file for the application (usually called server.js or app.js )
2. I had to rename .next to _next as assets urls were pointing to _next/

`const path = require("path");

const nextPath = path.join(__dirname, "node_modules", ".bin", "next");

process.argv.length = 1;
process.argv.push(nextPath, "start");

require(nextPath);
`
